packages/gatsby-plugin-thumbor/README.md

```diff
@@ -0,0 +1,123 @@
+
+As mentioned before, Thumbor has lots of different parameters that tell it how to process and convert images. `gatsby-plugin-thumbor` tightly integrates with `gatsby-plugin-image` to give you control over these parameters in a simple interface, while generating a thumbor compatible urls. The interface chosen is the extra `options` parameters in `gatsby-plugin-image` functions.
+
+## Securing your thumbor service
+You may now be wondering why there is an `/unsafe` segment in thumbor's generated url. This is because, thumbor accepts requests from anyone on the internet to process and resize images. This makes the thumbor service prone to DDoS attacks.
+ 
+Thumbor has a solution for this on their docs. They say you should sign the urls so the thumbor service know when an image is from a reliable source or if it's someone messing with your server. This is the best way for protecting your thumbor server. 
+
+However, this plugin offers an alternative to this solution. 
+Usually, when serving gatsby, you are using a CDN (`gatsby-plugin-netlify`) or nginx (`@vtex/gatsby-plugin-nginx`). These services can be used as a reverse proxy server. Adding a reverse proxy to your thumbor service may protect it from malicous users because then we can only allow a small subset of all possible transformations to your images, decreasing the attack space on your thumbor service. 
+This protection is achieved with two parameters accepted by the plugin, `sizes` and `basePath`. A config using these parameters look like this:
+```js
+// In your gatsby-config.js
+module.exports = {
+  plugins: [
+    // other plugins ...
+    {
+      resolve: 'gatsby-plugin-image'
+    },
+    {
+      resolve: `@vtex/gatsby-plugin-thumbor`
+      options: {
+        // Your thumbor server url
+        server: 'https://mythumborserver.com',
+        basePath: '/images',
+        sizes: [
+          '1080x1080',
+          '720x720',
+        ]
+      }
+    }
+  ],
+}
+```
+
+Now, instead of creating urls like `https://mythumborserver.com/unsafe/1080x1080/http://example.com/image.png`, the plugin will create urls for `{basePath}/1080x1080/http://example.com/image.png`, serve thumbor images on the same domain as your site and will only allow resizing to 1080x1080 and 720x720 images.
+
+Behind the scenes, what this plugin is doing is calling a `createRedirect` function between `/{basePath}/{size} => https://mythumborserver.com/unsafe/{size}` 
+
+## Examples
+Aside from `useThumborImageData`, this plugin exports `useUrlBuilder`, `useGetThumborImageData`, `getThumborImageData` and `urlBuilder`. These functions are thin wrappers to their counterparts on `gatsby-plugin-image` only adding the plugin's options, like the thumbor server url. You can use them normally in your react code. For instance, if you need the `getImageData` function you can:
```

Comment:
```suggestion
Now, instead of creating URLs like `https://mythumborserver.com/unsafe/1080x1080/http://example.com/image.png`, the plugin will create URLs for `{basePath}/1080x1080/http://example.com/image.png`, serve thumbor images on the same domain as your site, and will only allow resizing to 1080x1080 and 720x720 images.

Behind the scenes, what this plugin is doing is calling a `createRedirect` function between `/{basePath}/{size} => https://mythumborserver.com/unsafe/{size}` 

## Examples
Aside from `useThumborImageData`, this plugin exports `useUrlBuilder`, `useGetThumborImageData`, `getThumborImageData` and `urlBuilder`. These functions are thin wrappers to their counterparts on `gatsby-plugin-image` only adding the plugin's options, like the thumbor server URL. You can use them normally in your react code. For instance, if you need the `getImageData` function you can:
```